### PR TITLE
secret pkg: WithAnnotations and Get methods added + unit test

### PR DESCRIPTION
**Pull request by elenagerman**
secret pkg: WithAnnotations and Get methods added + unit test

### CODE REVIEW

**Comment by elenagerman** [comment]
@kononovn @trewest @cdvultur @achuzhoy PTAL

File: pkg/clients/clients.go [hunk 1]
```diff
@@ -441,6 +441,8 @@ func GetTestClients(tcp TestClientParams) *Settings {
 		// Add more items below if/when needed.
 		switch v.(type) {
 		// K8s Client Objects
+		case *corev1.Secret:
```
Comment (kononovn) [inline pkg/clients/clients.go:444]: Please remove it. We'll implement secret unit test in different PR

File: pkg/secret/secret.go [hunk 2]
```diff
@@ -90,6 +106,26 @@ func Pull(apiClient *clients.Settings, name, nsname string) (*Builder, error) {
 	return &builder, nil
 }
 
+// Get fetches the defined secret from the cluster.
+func (builder *Builder) Get() (*corev1.Secret, error) {
```
Comment (kononovn) [inline pkg/secret/secret.go:110]: Please remove this function. It's not needed for standard k8s objects. For standard k8s objects we have full client interface support and we use Get method directly in Exist.

**Review by kononovn — CHANGES_REQUESTED** [review]
2 comments to fix

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**